### DISCUSSION TITLE
✨ Observability - Find Opened/Read/Touched Files

### DISCUSSION THREAD

**Discussion by stobias123** [dagger/dagger#12507] [Ideas]
### What are you trying to do?

Would like to get a list of which files were read/opened during a pipeline run. 

Context in discord [here](https://discord.com/channels/707636530424053791/708371226174685314/1100126544963305592)

### Why is this important to you?

Lots of boilerplate code in our repos, significant amounts aren't actually used. Would be good cool to be able to say "xyz files can likely be deprecated"

### How are you currently working around this?

If this isn't possible directly through buildkit or dagger, I was thinking it could be implemented somehow via epbf probes.

**Comment by marcosnils** [comment]
I'd also add to this that having better visibility about which files are being sent to the context on each build could also help to stop some performance improvements of unnecessary file transfers.
